File: java/mediator/exercise/CommercialAirplane.java
package mediator.exercise;

public class CommercialAirplane implements Airplane {
    private AirTrafficControlMediator mediator;

    public CommercialAirplane(AirTrafficControlMediator mediator) {
        this.mediator = mediator;
    }

    @Override
    public String getType() {
        return "Commercial";
    }

    @Override
    public void requestTakeoff() {
        mediator.requestTakeOff(this);
    }

    @Override
    public void requestLanding() {
        mediator.requestLanding(this);
    }

    @Override
    public void notifyAirTrafficControl(String message) {
        System.out.println("Commercial Airplane: " + message);
    }
}

File: C#/Mediator/Exercise/CommercialAirplane.cs
namespace Mediator.Exercise;

public sealed class CommercialAirplane : IAirplane
{
    public string Type { get; } = "Commercial";

    private readonly IAirTrafficControlMediator _airTrafficControlMediator;

    public CommercialAirplane(IAirTrafficControlMediator airTrafficControl)
        => _airTrafficControlMediator = airTrafficControl;

    public void NotifyAirTrafficControl(string message)
        => Console.WriteLine($"Commercial Airplane: {message}");

    public void RequestLanding()
        => _airTrafficControlMediator.RequestLanding(this);

    public void RequestTakeoff()
        => _airTrafficControlMediator.RequestTakeOff(this);
}
